offchain-modules/packages/x/src/handlers/eth.ts

```diff
@@ -465,7 +718,7 @@ export class EthHandler {
   start(): void {
     void this.watchNewBlock();
 
-    this.handleUnlockRecords();
+    Promise.all([this.handleUnlockRecords, this.handleMintRecords]).catch((e) => logger.error(e));
```

Comment:
Cann't do unlock and mint concurrently

Reply:
> ```ts
>   handleTodoMintAndUnlockRecords(): void {
>     foreverPromise(
>       async () => {
>         await this.todoMintRecordsHandler();
>         await this.todoUnlockRecordsHandler();
>       },
>       {
>         onRejectedInterval: 30000,
>         onResolvedInterval: 30000,
>         onRejected: (e: Error) => {
>           logger.error(`error:${e.stack}`);
>         },
>       },
>     );
>   }
> ```
> 
> await tasks in one foreverPromise can be executed sequentially

This code is outdated. I just write a new task scheduler to manage the handlers.

Reply:
> ```ts
>   handleTodoMintAndUnlockRecords(): void {
>     foreverPromise(
>       async () => {
>         await this.todoMintRecordsHandler();
>         await this.todoUnlockRecordsHandler();
>       },
>       {
>         onRejectedInterval: 30000,
>         onResolvedInterval: 30000,
>         onRejected: (e: Error) => {
>           logger.error(`error:${e.stack}`);
>         },
>       },
>     );
>   }
> ```
> 
> await tasks in one foreverPromise can be executed sequentially

If the first handler meet an error while executing, the second one will not execute until the error goes away.